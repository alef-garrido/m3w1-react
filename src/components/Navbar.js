import React from 'react';
import { Link } from 'react-router-dom';

export default function Navbar() {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <nav className="navbar">
      <h1>Math Magicians</h1>
      <ul>
        {links.map((link) => (
          <li key={link.id} className="nav-item">
            <Link to={link.path}>
              {link.text}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
}
