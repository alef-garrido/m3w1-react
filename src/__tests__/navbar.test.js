import { BrowserRouter as Router } from "react-router-dom";
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar'

it('renders as expected', () => {
  const tree = renderer 
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();      
}) 