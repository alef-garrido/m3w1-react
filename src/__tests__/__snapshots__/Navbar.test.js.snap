// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<nav
  className="navbar--container"
>
  <div
    className="navbar--logo"
  >
    <h2>
      Math4Ever
    </h2>
    <div
      className="navbar--deco"
    />
  </div>
  <ul>
    <li
      className="nav-item"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
